src/irmin-pack/unix/s.ml

```diff
@@ -40,6 +40,15 @@ module type S = sig
   (** [flush t] flush read-write pack on disk. Raises [RO_Not_Allowed] if called
       by a readonly instance.*)
 
+  type snap = {
+    generation : int;
+    suffix_start_offset : int63;
+    prefix_commit_offset : int63;
+  }
+
+  val export : repo -> commit_key -> snap Lwt.t
```

Comment:
Following an offline discussion with @icristescu, a good story for that export function could be that export takes the path where a directory will be created containing a fully functional irmin-pack store which contains only the commit (and it's tree).

This way, no import function is needed.

Reply:
Would it make sense to separate the `export` from running the GC? (= copy the existing prefix/mapping with `first_commit` as `commit_key`, without running a (destructive) GC to build the snapshot)

Reply:
IIUC, this code runs the worker part of the normal GC but doesn't run the finalise part of the normal GC. 

The worker part only creates new files, it doesn't modify the existing files in any way.

Does this answer your question?